File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode t=head;
        int c=0;
        while(t!=null){
            c++;
            t=t.next;
        }
        int pos=c-n;
        System.out.print(pos);
        if(pos==0){
            head=head.next;
            return head;
        }
        t=head;
        for(int i=1;i<pos;i++){
            t=t.next;
        }
        t.next=t.next.next;
        return head;
    }
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* st=new ListNode(0);
        st->next=head;
        ListNode* f=st;
        ListNode* s=st;
        for(int i=0;i<=n;i++){
            f=f->next;
        }
        while(f!=nullptr){
            f=f->next;
            s=s->next;
        }
        s->next=s->next->next;
        return st->next;
    }
};